maintenance/wikia/WikiaMaps/removeImapTags.php

```diff
@@ -0,0 +1,223 @@
+		$this->addOption( 'maps-db-host', 'WikiaMaps DB host', $required = true, $withArg = true );
+		$this->addOption( 'maps-db-user', 'WikiaMaps DB user', $required = true, $withArg = true );
+		$this->addOption( 'maps-db-pass', 'WikiaMaps DB pass', $required = true, $withArg = true );
+		$this->addOption( 'maps-db-name', 'WikiaMaps DB name', $required = true, $withArg = true );
+		$this->addOption(
+			'tiles-set-id',
+			'Specify a tiles\' set id which leads to a group of maps which <imap /> tags should get removed',
+			$required = true,
+			$withArg = true,
+			'tsi'
+		);
+	}
+
+	static public function isTest() {
+		return self::$test;
+	}
+
+	static public function isNaturalNumber( $int ) {
+		return intval( $int ) > 0;
+	}
+
+	public function getMapsIdsUsingTileset( $cityId, $tilesSetId ) {
+		$dbw = new DatabaseMysqli(
+			$this->getOption('maps-db-host'),
+			$this->getOption('maps-db-user'),
+			$this->getOption('maps-db-pass'),
+			$this->getOption('maps-db-name')
+		);
+
+		return $dbw->selectFieldValues( 'map', 'id', [
+				'city_id' => $cityId,
+				'tile_set_id' => $tilesSetId
+		] );
+	}
+
+	public function getArticlesIdsUsingImap( $cityId ) {
+		global $wgStatsDB;
+
+		$dbw = wfGetDB(DB_SLAVE, array(), $wgStatsDB);
+		return $dbw->selectFieldValues( 'city_used_tags', 'ct_page_id', [
+			'ct_wikia_id' => $cityId,
+			'ct_kind' => 'imap'
+		] );
+	}
+
+	public function hasImapTag( $articleId, &$foundTags, &$foundTagsMapIds ) {
+		$article = Article::newFromID( $articleId );
+
+		if ( $article instanceof Article && $article->getID() ) {
+			$results = null;
+			$articleContent = $article->getContent();
+			$imapSearchRegexp = "/<imap.*map\-id\=['\"](\d{1,})['\"].*(<\/imap|\/)>/";
+
+			$noOfFoundTags = preg_match_all($imapSearchRegexp, $articleContent, $results);
+
+			if( $noOfFoundTags === 0 ) {
+				return false;
+			}
+
+			$foundTags = $results[0];
+			$foundTagsMapIds = $results[1];
+
+			return true;
+		} else {
+			$this->output( sprintf( 'Article #%d does not exist anymore', $articleId ) . PHP_EOL );
+		}
+	}
+
+	public function hasTagsToRemove( $foundTagsMapIds, $mapsUsingTheTileset, &$tagsToRemoveArrayKeyIds ) {
+		foreach( $foundTagsMapIds as $key => $mapId ) {
+			if ( in_array( $mapId, $mapsUsingTheTileset ) ) {
+				$tagsToRemoveArrayKeyIds[] = $key;
+			}
+		}
+
+		return !empty( $tagsToRemoveArrayKeyIds );
+	}
+
+	public function removeImapTagFromArticle( $articleId, $stringWithTagToRemove ) {
+		$user = User::newFromName( 'WikiaBot' );
+		$article = Article::newFromID( $articleId );
+
+		if ( $user->getId() === 0 ) {
+			$this->output( 'Failed using WikiaBot user' . PHP_EOL );
+		}
+
+		if ( $article instanceof Article && $article->getID() ) {
+			$oldContent = $article->getContent();
+			$newContent = str_replace( $stringWithTagToRemove, "", $oldContent );
+
+			$this->output( sprintf( 'Trying to edit article #%d', $articleId ) . PHP_EOL );
+			if( !self::isTest() ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)